reactive/kotlinx-coroutines-rx2/src/RxScheduler.kt

```diff
@@ -4,17 +4,84 @@
 
 package kotlinx.coroutines.rx2
 
-import io.reactivex.Scheduler
+import io.reactivex.*
+import io.reactivex.disposables.*
+import io.reactivex.plugins.*
 import kotlinx.coroutines.*
-import java.util.concurrent.TimeUnit
-import kotlin.coroutines.CoroutineContext
+import java.util.concurrent.*
+import kotlin.coroutines.*
 
 /**
  * Converts an instance of [Scheduler] to an implementation of [CoroutineDispatcher]
  * and provides native support of [delay] and [withTimeout].
  */
 public fun Scheduler.asCoroutineDispatcher(): SchedulerCoroutineDispatcher = SchedulerCoroutineDispatcher(this)
 
+/**
+ * Converts an instance of [CoroutineDispatcher] to an implementation of [Scheduler].
+ */
+public fun CoroutineDispatcher.asScheduler(): Scheduler =
+    if (this is SchedulerCoroutineDispatcher) {
+        scheduler
+    } else {
+        DispatcherScheduler(this)
+    }
+
+private class DispatcherScheduler(private val dispatcher: CoroutineDispatcher) : Scheduler() {
+
+    override fun scheduleDirect(run: java.lang.Runnable): Disposable {
+        val decoratedRun = RxJavaPlugins.onSchedule(run)
+        val worker = createWorker() as DispatcherWorker
+        worker.schedule(decoratedRun)
+        return worker
+    }
+
+    override fun scheduleDirect(run: java.lang.Runnable, delay: Long, unit: TimeUnit): Disposable {
+        val decoratedRun = RxJavaPlugins.onSchedule(run)
+        val worker = createWorker() as DispatcherWorker
+        worker.schedule(decoratedRun, delay, unit)
+        return worker
+    }
+
+    private class DispatcherWorker(private val dispatcher: CoroutineDispatcher) : Worker() {
+
+        private val workerScope = CoroutineScope(SupervisorJob())
+
+        override fun isDisposed(): Boolean = !workerScope.isActive
+
+        override fun schedule(run: java.lang.Runnable): Disposable {
+            return if (workerScope.isActive) {
```

Comment:
Please expression function form (`= if ...`) if the function is doing just `return`.